make to_edge_transform_and_lower support  etrecord generation

Gasoonjia opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #13058
* #13052
* __->__ #13034
* #13020
* #13010

Differential Revision: [D79336982](https://our.internmc.facebook.com/intern/diff/D79336982/)

umbrella issue: https://github.com/pytorch/executorch/issues/12961

Gasoonjia committed:
make to_edge_transform_and_lower support  etrecord generation



Differential Revision: [D79336982](https://our.internmc.facebook.com/intern/diff/D79336982/)

[ghstack-poisoned]

Gasoonjia committed:
Update on "make to_edge_transform_and_lower support  etrecord generation"


Differential Revision: [D79336982](https://our.internmc.facebook.com/intern/diff/D79336982/)

umbrella issue: https://github.com/pytorch/executorch/issues/12961

[ghstack-poisoned]

Gasoonjia committed:
Update on "make to_edge_transform_and_lower support  etrecord generation"


Differential Revision: [D79336982](https://our.internmc.facebook.com/intern/diff/D79336982/)

umbrella issue: https://github.com/pytorch/executorch/issues/12961

[ghstack-poisoned]

GregoryComer commented:
Can you add a test that covers to_edge_transform_and_lower with a partitioner? It might already handle this case, but to_backend creates a new edge program manager and is only called internally when a partitioner is provided. It might need some extra logic for this case.

Gasoonjia committed:
Update on "make to_edge_transform_and_lower support  etrecord generation"


Differential Revision: [D79336982](https://our.internmc.facebook.com/intern/diff/D79336982/)

umbrella issue: https://github.com/pytorch/executorch/issues/12961

[ghstack-poisoned]

Gasoonjia committed:
Update on "make to_edge_transform_and_lower support  etrecord generation"


Differential Revision: [D79336982](https://our.internmc.facebook.com/intern/diff/D79336982/)

umbrella issue: https://github.com/pytorch/executorch/issues/12961

[ghstack-poisoned]
